packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt

```diff
@@ -341,6 +345,24 @@ class VideoModule : Module() {
           ref.player.play()
         }
       }
+
+      AsyncFunction("generateThumbnailsAsync") Coroutine { ref: VideoPlayer, times: List<Duration> ->
+        return@Coroutine ref.toMetadataRetriever().safeUse {
+          val bitmaps = times.map { time ->
+            appContext.backgroundCoroutineScope.async {
+              generateThumbnailAtTime(time)
+            }
+          }
+
+          bitmaps.awaitAll()
+        }
+      }
+
+      Class<VideoThumbnail> {
+        Property("width") { ref -> ref.width }
+        Property("height") { ref -> ref.height }
+        Property("requestedTime") { ref -> ref.requestedTime }
```

Comment:
If there is no way to get the `actualTime`, we should make it optional (or nullable) in TypeScript. Could we estimate it maybe? Does Android always use the nearest frame that you request for or has some type of threshold like iOS?

Reply:
I managed to calculate the frame time. It might not always be available, but I didn't find a better way of calculating it.